### DISCUSSION TITLE
Handling types for different schemas

### DISCUSSION THREAD

**Discussion by presedo93** [supabase/supabase#15138] [Questions]
Hi! 

I've been trying for a while but I haven't reached yet to a solution. I have my DB structured in several schemas: users, plans, public, etc... and when I generate the types for typescript I follow the steps on the documentation:

- I add in the `config.toml` the two schemas to the `api` section.
- I generate the types using `supabase gen types typescript --local > ./src/types/schema.ts`.

Finally, I import the `Database` and use as the docs suggest:

```TS
import { Database } from '@/types/schema'
import { createClient } from '@supabase/supabase-js'
import AsyncStorage from '@react-native-async-storage/async-storage'

export const supabase = createClient<Database>(supabaseUrl, supabaseAnonKey, {
  auth: {
    storage: AsyncStorage as any,
    autoRefreshToken: true,
    persistSession: true,
    detectSessionInUrl: false
  }
})
```
The issue comes when I try to get the type hints on methods like `from`. I can't get them.

If I generate the types using the `--schema users`, it will work for the `users` schema (same thing for other schemas) but would that mean that I would have to create clients for every schema that I import?

I also considered this:

```
export type Workouts = Database['workouts']
export type Users = Database['users']

supabase = createClient<Database>(...)

await supabase.from<Users>(...)
```

How could I handle several schemas and type hints?

Thanks!!

**Comment by JungeWerther** [comment]
You have to add `db` parameter like so:

```
const client = createClient(
    supabaseUrl, 
    supabaseKey, {
      db: {
        schema: 'my_custom_schema_name'
      }
    })
```

**Comment by lorikku** [comment]
Hi just leaving this here for anyone who needs the answer: https://github.com/orgs/supabase/discussions/32841
